rotkehlchen/db/dbhandler.py

```diff
@@ -2772,27 +2768,50 @@ def get_latest_asset_value_distribution(self) -> List[DBAssetBalance]:
         """
         with self.conn.read_ctx() as cursor:
             ignored_assets = self.get_ignored_assets(cursor)
+            treat_eth2_as_eth = self.get_settings(cursor).treat_eth2_as_eth
             cursor.execute(
                 'SELECT time, currency, amount, usd_value, category FROM timed_balances WHERE '
                 'time=(SELECT MAX(time) from timed_balances) AND category = ? ORDER BY '
                 'CAST(usd_value AS REAL) DESC;',
                 (BalanceType.ASSET.serialize_for_db(),),  # pylint: disable=no-member
             )
             asset_balances = []
+            eth_balance = DBAssetBalance(
+                time=Timestamp(0),
+                category=BalanceType.ASSET,
+                asset=A_ETH,
+                amount=ZERO,
+                usd_value=ZERO,
+            )
             for result in cursor:
                 asset = Asset(result[1])
+                time = Timestamp(result[0])
+                amount = FVal(result[2])
+                usd_value = FVal(result[3])
                 if asset in ignored_assets:
                     continue
-                asset_balances.append(
-                    DBAssetBalance(
-                        time=result[0],
-                        asset=asset,
-                        amount=result[2],
-                        usd_value=result[3],
-                        category=BalanceType.deserialize_from_db(result[4]),
-                    ),
-                )
-
+                # show eth & eth2 as eth in value distribution by asset
+                if treat_eth2_as_eth is True and asset in (A_ETH, A_ETH2):
+                    eth_balance.time = time
+                    eth_balance.amount = eth_balance.amount + amount
+                    eth_balance.usd_value = eth_balance.usd_value + usd_value
+                else:
+                    asset_balances.append(
+                        DBAssetBalance(
+                            time=time,
+                            asset=asset,
+                            amount=amount,
+                            usd_value=usd_value,
+                            category=BalanceType.deserialize_from_db(result[4]),
+                        ),
+                    )
+            # only add the eth_balance if it contains a balance > 0
+            if eth_balance.amount > ZERO or eth_balance.usd_value > ZERO:
```

Comment:
no need for both checks

```suggestion
            if eth_balance.amount > ZERO:
```